h3/src/quic.rs

```diff
@@ -75,6 +120,15 @@ pub trait Connection<B: Buf> {
 
     /// Close the connection immediately
     fn close(&mut self, code: crate::error::Code, reason: &[u8]);
+
+    /// Poll the connection for incoming datagrams.
+    fn poll_accept_datagram(
```

Comment:
Are these actions "required" for h3 and quic? Or is it an extension, that some implementations might not bother to do? If it's optional, how should we allow people to not implement it?

Maybe as a separate trait, that if also implemented, then people can access the methods on `h3::client` or `h3::server` to deal with datagrams?

Reply:
It is an HTTP/3 extension.

I could quite easily split it into a trait inside of h3/quic, if so desired.

https://www.rfc-editor.org/rfc/rfc9297

Reply:
I have now split it into a `SendDatagramExt` and `RecvDatagramExt` that use the same "buffer specialization" pattern we've talked about before. Looks a lot cleaner now in my opinion.